cmd/dex/serve.go

```diff
@@ -175,6 +147,50 @@ func serve(cmd *cobra.Command, args []string) error {
 		}
 		s = storage.WithStaticPasswords(s, passwords)
 	}
+	if len(c.StaticConnectors) > 0 {
+		connectors := make([]storage.Connector, len(c.StaticConnectors))
+		for i, c := range c.StaticConnectors {
+			if c.ID == "" || c.Name == "" || c.Type == "" {
+				return fmt.Errorf("invalid config: ID, Type and Name fields are required for a connector")
+			}
+			if c.Config == nil {
+				return fmt.Errorf("invalid config: no config field for connector %q", c.ID)
+			}
+			logger.Infof("config connector: %s", c.ID)
+			conn, err := ToStorageConnector(c, "1")
+			if err != nil {
+				return fmt.Errorf("failed to initialize connectors: %v", err)
+			}
+			connectors[i] = conn
+		}
+		s = storage.WithStaticConnectors(s, connectors)
+	}
+
+	// Retrieves connector objects in backend storage. This list includes the static connectors
+	// defined in the ConfigMap and dynamic connectors retrieved from the storage.
+	storageConnectors, err := s.ListConnectors()
```

Comment:
This seems like this should be something done by the server. cmd is just for validating inputs, not checking existing storage values.